src/routes/comments.js

```diff
@@ -0,0 +1,122 @@
+const express = require("express");
+const router = express.Router();
+const { body, validationResult } = require("express-validator");
+const Comment = require("../models/Comment");
+const User = require("../models/User");
+const mongoose = require("mongoose");
+
+//Comment Routes
+
+/*
+Route GET requests at root 
+return response containing all comments in JSON on success
+*/
+router.get("/", function(req, res) {
+  Comment.find()
+    .lean()
+    .exec(function(err, comments) {
+      if (err) {
+        {
+          return res.status(500).send(err);
+        }
+      } else {
+        return res.json(comments);
+      }
+    });
+});
+
+/*
+Route Post requests 
+Validate request body
+Verify userID exists in db
+Create new comment document from request body
+Populate username of comment document
+Return response containing new comment in JSON
+*/
+
+router.post(
+  "/",
+  [
+    body("userID", "invalid userID").isLength({
+      min: 24
+    }),
+    body("text", "Comment is empty")
+      .not()
+      .isEmpty()
+  ],
+  async (req, res) => {
+    //express-validator
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    const { userID, text, date } = req.body;
+
+    const _id = new mongoose.Types.ObjectId(userID);
+
+    try {
+      const user = await User.findOne({ _id });
+      if (!user) {
+        return res.status(404).json({ error: "No user Found" });
+      }
+
+      let new_comment = await Comment.create({
+        userID,
+        text,
+        date
+      });
+
+      new_comment = await new_comment
+        .populate("username", userID.username)
```

Comment:
I have a little confused in this part. Can you explain a little more details here?

Reply:
Yes, I can explain.  This routes post requests for comments. The second argument is an array that validates the contents of the request body, using an express-validator. The userID should be 24 characters in length and the text field should not have an empty value. The function returns a status error and array of errors generated by express-validator.

After confirming the request body is valid, I create a new mongoose ObjectID from userID. I confirm the user exists by passing the ObjectID into findOne(). If the user doesn't exist, I return 404 status error and an error message.

If the user exists, I create a new comment using the variables created by destructuring req.body. Once the new comment is created, I populate the username field using userID.

I think I should use the mongoose ObjectID instead of userID when creating the comment:

 ` let new_comment = await Comment.create({
        userID: _id,
        text,
        date
      });
` 
The userID is a string, the mongoose ObjectID is an object. The string is converted into an object when used in find() queries, so I just wanted to be consistent and store the IDs as Objects instead of strings.

Reply:
In fact I originally referred to the odes below because this is not my normal way to do it. Anyway I'll pull your branch and test it.
 new_comment = await new_comment
        .populate("username", userID.username)
        .execPopulate();

Reply:
I just tested to delete the code below:

new_comment = await new_comment
.populate("username", userID.username)
.execPopulate();

the new_comment still show the same result with or without it.